crypto/s2n_hash.c

```diff
@@ -562,26 +566,26 @@ int s2n_hash_init(struct s2n_hash_state *state, s2n_hash_algorithm alg)
 
 int s2n_hash_update(struct s2n_hash_state *state, const void *data, uint32_t size)
 {
-    PRECONDITION_POSIX(s2n_hash_state_is_valid(state));
-    PRECONDITION_POSIX(S2N_MEM_IS_READABLE(data, size));
+    PRECONDITION_POSIX(s2n_hash_state_validate(state));
+    ENSURE_POSIX(S2N_MEM_IS_READABLE(data, size), S2N_ERR_PRECONDITION_VIOLATION);
```

Comment:
Why are these different?

Reply:
`PRECONDITION_POSIX` now accepts `s2n_result` instead of Boolean values, that's why we need to use `ENSURE_POSIX`when checking for specific preconditions now.

Reply:
I'm gonna update these macros. Tracking your request in issue https://github.com/awslabs/s2n/issues/2461.